tests/components/altruist/test_config_flow.py

```diff
@@ -0,0 +1,194 @@
+"""Test the Altruist Sensor config flow."""
+
+from ipaddress import ip_address
+from unittest.mock import patch
+
+from altruistclient import AltruistError
+
+from homeassistant import config_entries
+from homeassistant.components.altruist.const import (
+    CONF_DEVICE_ID,
+    CONF_IP_ADDRESS,
+    DOMAIN,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+from homeassistant.helpers.service_info.zeroconf import ZeroconfServiceInfo
+
+from tests.common import MockConfigEntry
+
+ZEROCONF_DISCOVERY = ZeroconfServiceInfo(
+    ip_address=ip_address("192.168.1.100"),
+    ip_addresses=[ip_address("192.168.1.100")],
+    hostname="altruist-purple.local.",
+    name="altruist-purple._altruist._tcp.local.",
+    port=80,
+    type="_altruist._tcp.local.",
+    properties={
+        "PATH": "/config",
+    },
+)
+
+
+async def test_form_user_step(hass: HomeAssistant) -> None:
+    """Test we get the form for user step."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+
+async def test_form_user_step_invalid_ip(hass: HomeAssistant) -> None:
+    """Test we handle invalid IP address."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {CONF_IP_ADDRESS: "invalid_ip"},
+    )
+
+    assert result2["type"] is FlowResultType.FORM
+    assert result2["errors"] == {"base": "invalid_ip"}
+
+
+async def test_form_user_step_cannot_connect_then_recovers(
+    hass: HomeAssistant, mock_altruist_client
```

Comment:
```suggestion
    hass: HomeAssistant, mock_altruist_client: MagicMock
```
is missing its type everywhere